### PR TITLE
My submission js2

### PR DESCRIPTION
**Pull request by Ousman37**
Please can you review My social media app auth? 

Thanks:)

### CODE REVIEW

**Comment by Fermain** [comment]
There are 5000 changes, so I am unable to review at this stage. Please remove node_modules from your history and gitignore this at all times.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/node_modules
```
Comment (Fermain) [inline .gitignore]: You need to delete node_modules from history before adding this, comment it out, then delete the folder, commit that, then add this back again.

Comment (Ousman37) [reply .gitignore]: OK, let go and delete it then.

**Comment by Ousman37** [comment]
I have deleted it and added a new one again.

 Please let me know if it is fine now.

**Comment by Fermain** [comment]
Looks good now, will review shortly

**Comment by Ousman37** [comment]
Thank you:)  I had it in my previous set-up. Everything was fine, but then I have trouble with my commit in the terminal.  I try the way you showed us working on three different branches, and it did not work for me. I had some conflicts with the branches. 

After this, I will practice more with small branches to make a commit different branches

**Comment by Ousman37** [comment]
Hello man, 

Hope you did not forget me here. 

Have a great weekend out there!

File: server.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-undef */
+const express = require('express');
```
Comment (Fermain) [inline server.js:2]: Are you using express instead of live server?

File: src/create-post.html [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            <input id="tags" name="Tags" required type="text" class="form-control" placeholder="Tags" />
+
+          </div>
+
+          <div class="form-group">
+            <label for="media">Media</label>
+            <input id="media" name="Media" required type="text" class="form-control"
+              placeholder="Enter a image link for Banner" />
+          </div>
+          <button id="create-submit" type="button" class="btn btn-primary">Create Entry</button>
```
Comment (Fermain) [inline src/create-post.html]:
```suggestion
          <button id="create-submit" class="btn btn-primary">Create Entry</button>
```

Type button doesn't trigger a submit, have you tested this?

File: src/edit.html [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+          </div>
+
+          <div class="form-group">
+            <label for="media">Media</label>
+            <input id="media" name="Media" required type="text" class="form-control"
+              placeholder="Enter a image link for Banner" />
+
+          </div>
+          <button id="edit-submit" type="button" class="btn btn-primary">Save</button>
```
Comment (Fermain) [inline src/edit.html]:
```suggestion
          <button id="edit-submit" class="btn btn-primary">Save</button>
```

File: src/login.html [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+              <input title=" Must be at least 8 characters" pattern="[a-zA-Z0-9]{8,}" class="form-control login"
+                type="text" id="password" name="password" placeholder="qwerldskfjls23" required />
+
+              <ul class="helper-text">
+                <li class="length">Must be at least 8 characters long.</li>
+                <li class="lowercase">Must contain a lowercase letter.</li>
+                <li class="uppercase">Must contain an uppercase letter.</li>
+                <li class="special">Must contain a number or special character.</li>
+              </ul>
+              <button type="button" id="login-submit"
```
Comment (Fermain) [inline src/login.html]:
```suggestion
              <button id="login-submit"
```

File: src/register.html [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+
+          <div class="form-group">
+            <label for="password">Password</label>
+            <input id="password" name="password" pattern="[a-zA-Z0-9]{8,}" class="form-control"
+              required placeholder="password" />
+            <p class="alert alert-danger"> Must be at least 8 characters</p>
+          </div>
+
+          <button type="button" id="register-submit" class="btn_fourth">Register</button>
```
Comment (Fermain) [inline src/register.html]:
```suggestion
          <button id="register-submit" class="btn_fourth">Register</button>
```

**Comment by Fermain** [comment]
I can't see a way to create a post in the UI

File: server.js (hunk 2 above)
Comment (Ousman37) [reply server.js:2]: Yes, I am using express is okay?

**Comment by Ousman37** [comment]
I think I know why you couldn't see create a post in the UI I am going to fix that.

File: server.js (hunk 2 above)
Comment (Fermain) [reply server.js:2]: Yeah its ok

File: src/create-post.html (hunk 3 above)
Comment (Ousman37) [reply src/create-post.html]: Yes I have tested it and it works with the type it doesn´t trigger I have Tried your suggestion, but then I couldn't log in.

**Comment by Ousman37** [comment]
I have fixed the issue with the create a post in the UI. You will able to see it now. 

let me know.

Thanks :)

**Comment by Ousman37** [comment]
Hello man,  
I fixed the issue with the Buttons now all submit buttons are formatted correctly.  I guess I found what I was doing wrong in the  HTML form validation have tested it it works properly. 
 
Please let me know if there are any future changes  I could do will be glad to do them. 
Thanks :) 
Ethmane

**Comment by Fermain** [comment]
I'm testing now:

- [x] Delete Post (works but not clear to the user)
- [x] Update Post?
- [x] Profile?

**Comment by Ousman37** [comment]
Missing that part. Thank you for pointing that out. 
I’m going to work on that now. luckily, I still have  some times

**Comment by Ousman37** [comment]
Hello man, 

 I have updated the update post page the profile page is now updated. 
The deleted post  now gives the user a "message" that the post  has been  detected 
I have tested things are working fine. 

 Please let me know in case I'm blind by something. 

Thank you :)

**Comment by Fermain** [comment]
OK, I think you are just down to UX improvements now. Required functionality is in place but life as a user is not very easy. To improve this:

- Add headings to pages to make it clear what each page is for
- Make sure your link addresses are 100%, I got into a pickle by clicking through the links
- Redirect users after a successful delete or update

**Comment by Ousman37** [comment]
Thank you, man 😎

 I am going to fix that.

**Comment by Ousman37** [comment]
I have improved the UI / UX  now there’s a clear message appealing on the page for the user when they delete the post.

**Comment by Fermain** [comment]
Did you get my Loom feedback? There is one conflict to be resolved still before we can merge this in

**Comment by Ousman37** [comment]
Can you please check if it is fine now?

**Comment by Fermain** [comment]
This will go away when it's fine: 
image

**Pull request merged**